library/events/json_message_event.nim

```diff
@@ -0,0 +1,42 @@
+
+import
+    std/json
+import
+    ../../waku/v2/waku_core/message/message,
+    json_signal_event
+
+type JsonMessage = ref object
+  # https://rfc.vac.dev/spec/36/#jsonmessage-type
+  payload: string
+  contentTopic: string
+  version: uint
+  timestamp: int64
+
+type JsonMessageEvent* = ref object of JsonSignal
+    pubsubTopic*: string
+    messageId*: string
+    wakuMessage*: JsonMessage
+
+proc new*(T: type JsonMessageEvent,
+          pubSubTopic: string,
+          msg: WakuMessage): T =
+  # Returns a WakuMessage event as indicated in
+  # https://rfc.vac.dev/spec/36/#jsonmessageevent-type
+
+  var payload = newString(len(msg.payload))
+  copyMem(addr payload[0], unsafeAddr msg.payload[0], len(msg.payload))
+
+  return JsonMessageEvent(
+    eventType: "message",
+    pubSubTopic: pubSubTopic,
+    messageId: "TODO",
+    wakuMessage: JsonMessage(
+        payload: payload,
+        contentTopic: msg.contentTopic,
+        version: msg.version,
+        timestamp: int64(msg.timestamp)
```

Comment:
Some fields from WakuMessage are missing. Maybe it also makes sense to include the `proof` if RLN is enabled? 

What do you think if instead of creating a JsonMessage with the WakuMessage content, to instead serialize it with https://github.com/status-im/nim-json-serialization? I imagine it should be possible to deserialize it too in `waku_relay_publish`. The advantage of doing that is that if in the future new fields are added to `WakuMessage` protobuffer, they'll be serialized/unserialized automatically in the c-bindings API functions

Reply:
Yes, I absolutely agree. Nevertheless, I added the fields indicated in the next:
https://rfc.vac.dev/spec/36/#jsonmessage-type